components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -258,9 +259,32 @@ public static boolean isValidCaptcha(String reCaptchaResponse) throws CaptchaExc
         try {
             try (InputStream in = entity.getContent()) {
                 JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
-                if (verificationResponse == null || verificationResponse.get("success") == null ||
-                        !verificationResponse.get("success").getAsBoolean()) {
-                    throw new CaptchaClientException("reCaptcha verification failed. Please try again.");
+                if (verificationResponse == null) {
+                    throw new CaptchaClientException("Error receiving reCaptcha response from the server");
+                }
+                boolean success = verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS) != null
+                        && verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean();
```

Comment:
This can throw a ClassCastException if the value is not a valid boolean. Let's handle that

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/678